pkg/pipeline/source/web.go

```diff
@@ -262,81 +255,99 @@ func (s *WebSource) launchChrome(ctx context.Context, p *config.PipelineConfig,
 		chromedp.Flag("no-sandbox", !p.EnableChromeSandbox),
 	}
 
-	if insecure {
+	if p.Insecure {
 		opts = append(opts,
 			chromedp.Flag("disable-web-security", true),
 			chromedp.Flag("allow-running-insecure-content", true),
 		)
 	}
 
 	allocCtx, allocCancel := chromedp.NewExecAllocator(context.Background(), opts...)
-	chromeCtx, chromeCancel := chromedp.NewContext(allocCtx)
-	s.closeChrome = func() {
-		chromeCancel()
-		allocCancel()
-	}
+	s.closeChrome = allocCancel
 
-	chromedp.ListenTarget(chromeCtx, func(ev interface{}) {
-		switch ev := ev.(type) {
-		case *runtime.EventConsoleAPICalled:
-			for _, arg := range ev.Args {
-				var val interface{}
-				err := json.Unmarshal(arg.Value, &val)
-				if err != nil {
-					continue
-				}
+	for i := range chromeRetries {
+		if i > 0 {
+			logger.Debugw("navigation timed out, reloading")
+		}
 
-				switch fmt.Sprint(val) {
-				case startRecordingLog:
-					logger.Infow("chrome: START_RECORDING")
-					if s.startRecording != nil {
-						select {
-						case <-s.startRecording:
-							continue
-						default:
-							close(s.startRecording)
-						}
+		chromeCtx, chromeCancel := chromedp.NewContext(allocCtx)
```

Comment:
This is a pretty large code block inside a loop. Maybe deserves a separate function?

I cannot quickly tell if `chromeCancel` is always called right now, for example.